source/mir/random/discrete.d

```diff
@@ -109,6 +113,25 @@ unittest
     auto ds = discrete(cdPoints);
 
     auto obs = new uint[cdPoints.length];
-    foreach (i; 0..1000)
+    foreach (i; 0..10_000)
+        obs[ds(gen)]++;
+
+    assert(obs == [2536, 4963, 2501]);
+}
+
+// test with zero probabilities
+unittest
+{
+    import std.random : Random;
+    auto gen = Random(42);
```

Comment:
Agree, it's better to specify an exact generator just in case someone later adds some tests that depend on the pseudo-random sequence.